solidity/contracts/bridge/Bridge.sol

```diff
@@ -653,28 +653,42 @@ contract Bridge is Ownable {
         // all deposits.
         uint256 totalTreasuryFee = 0;
 
-        // Calculate the transaction fee per deposit by dividing the total
-        // transaction fee by deposits count. The total fee is just the
-        // difference between inputs amounts sum and the output amount.
-        // TODO: Deal with precision loss by having the last depositor pay
-        //       the higher transaction fee than others if there is a change,
-        //       just like it has been proposed in discussion:
-        //       https://github.com/keep-network/tbtc-v2/pull/128#discussion_r800555359.
-        uint256 txFee = (inputsInfo.inputsTotalValue - sweepTxOutputValue) /
-            inputsInfo.depositedAmounts.length;
-
-        require(txFee <= depositTxMaxFee, "Transaction fee is too high");
+        // Determine the transaction fee that should be incurred by each deposit
+        // and the indivisible remainder that should be additionally incurred
+        // by the last deposit.
+        (
+            uint256 depositTxFee,
+            uint256 depositTxFeeRemainder
+        ) = sweepTxFeeDistribution(
+                inputsInfo.inputsTotalValue,
+                sweepTxOutputValue,
+                inputsInfo.depositedAmounts.length
+            );
+
+        // Make sure the highest value of the deposit transaction fee does not
+        // exceed the maximum value limited by the governable parameter.
+        require(
+            depositTxFee + depositTxFeeRemainder <= depositTxMaxFee,
+            "Transaction fee is too high"
+        );
 
         // Reduce each deposit amount by treasury fee and transaction fee.
         for (uint256 i = 0; i < inputsInfo.depositedAmounts.length; i++) {
+            // The last deposit should incur the deposit transaction fee
+            // remainder.
+            uint256 depositTxFeeIncurred = i ==
+                inputsInfo.depositedAmounts.length - 1
+                ? depositTxFee + depositTxFeeRemainder
+                : depositTxFee;
```

Comment:
`depositTxFeeRemainder` will be at most `depositedAmounts.length - 1`.
If we assume the number of swept deposits is no greater than `10`, then `depositTxFeeRemainder` will be at most `9`.

So if we have 10 deposits, and tx fee is 999, the cost will spread like this:
99, 99, 99, 99, 99, 99, 99, 99, 99, 108.

9 satoshis paid more is not much and we can live with this.

Alternatively we could:
1) Force the wallet to always choose a fee that is divisible by the number of deposits
2) Or spread the cost more evenly:
  99 100 100 100 100 100 100 100 100 100

 So that the remainder is split evenly among the last depositors. 
 If the remainder is 9, add 1 to the cost of the last 9 depositors.
 If the remainder is 4, add 1 to the cost of the last 4 depositors.
 If the remainder is 1, add 1 to the cost of the last depositor.

Reply:
Yep, that's right, there are some on-chain alternatives. However, pushing the remainder to the last deposit seems to be the simplest and most gas-effective option at the moment. Currently, 9 satoshi is ~0.00354 USD so I think the additional cost for the last deposit is negligible. We can also implement the "divisible fee" behavior in the off-chain client at any moment.

Reply:
Although I do agree with @tomaszslabon that the two options he proposed are more fair, I also agree with @lukasz-zimnoch that we should choose the least expensive option given the gas consumption. Eventually, it is the depositor who will pay in the treasury fee for the gas consumption of the sweep proof transaction so option 2 could end up as theoretically more fair but practically, depositors would receive less value. Option 1 feels too dangerous to operators, in my opinion. The off-chain signing protocol will be complex enough even without additional constraints on the fee.